src/main/java/de/hysky/skyblocker/skyblock/profileviewer/utils/LevelFinder.java

```diff
@@ -0,0 +1,311 @@
+        boundaries.add(new LevelInfo(23950L, 19));
+        boundaries.add(new LevelInfo(30200L, 20));
+        boundaries.add(new LevelInfo(38050L, 21));
+        boundaries.add(new LevelInfo(47850L, 22));
+        boundaries.add(new LevelInfo(60100L, 23));
+        boundaries.add(new LevelInfo(75400L, 24));
+        boundaries.add(new LevelInfo(94500L, 25));
+        return boundaries;
+    }
+
+    private List<LevelInfo> createSocialSkillBoundaries() {
+        List<LevelInfo> boundaries = new ArrayList<>();
+        boundaries.add(new LevelInfo(0L, 0));
+        boundaries.add(new LevelInfo(50L, 1));
+        boundaries.add(new LevelInfo(150L, 2));
+        boundaries.add(new LevelInfo(300L, 3));
+        boundaries.add(new LevelInfo(550L, 4));
+        boundaries.add(new LevelInfo(1050L, 5));
+        boundaries.add(new LevelInfo(1800L, 6));
+        boundaries.add(new LevelInfo(2800L, 7));
+        boundaries.add(new LevelInfo(4050L, 8));
+        boundaries.add(new LevelInfo(5550L, 9));
+        boundaries.add(new LevelInfo(7550L, 10));
+        boundaries.add(new LevelInfo(10050L, 11));
+        boundaries.add(new LevelInfo(13050L, 12));
+        boundaries.add(new LevelInfo(16800L, 13));
+        boundaries.add(new LevelInfo(21300L, 14));
+        boundaries.add(new LevelInfo(27300L, 15));
+        boundaries.add(new LevelInfo(35300L, 16));
+        boundaries.add(new LevelInfo(45300L, 17));
+        boundaries.add(new LevelInfo(57800L, 18));
+        boundaries.add(new LevelInfo(72800L, 19));
+        boundaries.add(new LevelInfo(92800L, 20));
+        boundaries.add(new LevelInfo(117800L, 21));
+        boundaries.add(new LevelInfo(147800L, 22));
+        boundaries.add(new LevelInfo(182800L, 23));
+        boundaries.add(new LevelInfo(222800L, 24));
+        boundaries.add(new LevelInfo(272800L, 25));
+        return boundaries;
+    }
+
+    private List<LevelInfo> createGenericSlayerBoundaries() {
+        List<LevelInfo> boundaries = new ArrayList<>();
+        boundaries.add(new LevelInfo(0L, 0));
+        boundaries.add(new LevelInfo(5L, 1));
+        boundaries.add(new LevelInfo(15L, 2));
+        boundaries.add(new LevelInfo(200L, 3));
+        boundaries.add(new LevelInfo(1000L, 4));
+        boundaries.add(new LevelInfo(5000L, 5));
+        boundaries.add(new LevelInfo(20000L, 6));
+        boundaries.add(new LevelInfo(100000L, 7));
+        boundaries.add(new LevelInfo(400000L, 8));
+        boundaries.add(new LevelInfo(1000000L, 9));
+        return boundaries;
+    }
+
+    private List<LevelInfo> createVampireSlayerBoundaries() {
+        List<LevelInfo> boundaries = new ArrayList<>();
+        boundaries.add(new LevelInfo(0L, 0));
+        boundaries.add(new LevelInfo(20L, 1));
+        boundaries.add(new LevelInfo(75L, 2));
+        boundaries.add(new LevelInfo(240L, 3));
+        boundaries.add(new LevelInfo(840L, 4));
+        boundaries.add(new LevelInfo(2400L, 5));
+        return boundaries;
+    }
```

Comment:
There's gotta be formulas for this stuff.

Reply:
I looked it up and no, there is no formulas for this stuff

Reply:
I would shove that into other file and just look it up to not clog code

Reply:
I wonder, how other mods do this stuff?